Allow options to be passed into Ping

stblassitude opened the pull request:
This allows users to specify the broadcast address to send the ping to.
This is neccessary on systems with more than one interface, or certain
operating systems.

This allows KraigM/homebridge-harmonyhub#105 to be addressed.

swissmanu commented on lib/explorer.js:
```diff
@@ -54,13 +54,13 @@ function executeCleanUp () {
 }
 
-function Explorer (port) {
+function Explorer (port, options) {
```
i'd suggest to rename `options` to `pingOptions`. (two times `port` for example might be a bit confusing :) )

swissmanu commented on lib/ping.js:
```diff
@@ -1,13 +1,14 @@
 
-function Ping (portToAnnounce) {
+function Ping (portToAnnounce, options) {
   debug('Ping(' + portToAnnounce + ')')
```
what do you think about logging all new options here too?

swissmanu commented:
thank you for your addition @stblassitude! glad you can make use of `harmonyhubjs-discover` :)

after going through your change, may i kindly ask you to have a look at the following points?

* would you please have a look at my review comments?
* `harmonyhubjs-discover` is prepared for [conventional-changelog commits using commitizen](https://github.com/commitizen/cz-cli) which allows simplified/automated publish of new module versions. please rephrase your commits accordingly.

again, thank you for your contribution

swissmanu reviewed (changes requested):
thank you for your addition @stblassitude! glad you can make use of `harmonyhubjs-discover` :)

after going through your change, may i kindly ask you to have a look at the following points?

* would you please have a look at my review comments?
* `harmonyhubjs-discover` is prepared for [conventional-changelog commits using commitizen](https://github.com/commitizen/cz-cli) which allows simplified/automated publish of new module versions. please rephrase your commits accordingly.

again, thank you for your contribution

stblassitude commented:
Can I use Commitizen to rework this PR? Or do I need to close this and redo the commits on a fresh branch, and submit a fresh PR?

swissmanu commented:
i dont have any problem if you force-push your rebased changes to this PR ☺️

stblassitude committed:
feat(ping): Allow options to ping to set broadcast address

This allows users to specify the broadcast address to send the ping to.

This allows KraigM/homebridge-harmonyhub#105 to be addressed.

stblassitude commented:
Let me know if this is acceptable, or needs further refinement.

swissmanu commented:
perfect, thanks for reworking :)
will release a new version asap

swissmanu commented:
latest release [v1.1.1](https://github.com/swissmanu/harmonyhubjs-discover/releases/tag/v1.1.1) is available with your changes 🙌

stefanbethke commented:
Thank you!

stblassitude commented:
Thank you!
